### PR TITLE
fixes issue #3030 by checking for an empty field

### PR DESCRIPTION
**Pull request by JakeHL**
This PR fixes issue #3030 by checking for an empty directory when conditionally rendering the warning.

### CODE REVIEW

File: app/src/ui/add-repository/create-repository.tsx [hunk 1]
```diff
@@ -416,7 +416,7 @@ export class CreateRepository extends React.Component<
   private renderGitRepositoryWarning() {
     const isRepo = this.state.isRepository
 
-    if (this.state.path.length && !isRepo) {
+    if ((this.state.path.length && !isRepo) || this.state.path.length === 0) {
```
Comment (Daniel-McCarthy) [inline app/src/ui/add-repository/create-repository.tsx]: Thanks for the contribution. This seems to complicate the if condition somewhat. Why not something this below?
```typescript
if (this.state.path.length === 0 || !isRepo) {
```

Comment (JakeHL) [reply app/src/ui/add-repository/create-repository.tsx]: I completely didn't catch the string type not being nullable. I'll update the PR 👍

Comment (JakeHL) [reply app/src/ui/add-repository/create-repository.tsx]: I've updated the PR. Good catch on the non-nullable string. It basically removes any possible edge cases.

**Comment by a0930119952** [comment]
從 Windows 10 的郵件傳送

寄件者: Jake Langford
傳送時間: 2018年10月6日 上午 08:51
主旨: Re: [desktop/desktop] fixes issue #3030 by checking for an emptyfield (#5828)

@JakeHL commented on this pull request.

In app/src/ui/add-repository/create-repository.tsx:
> @@ -416,7 +416,7 @@ export class CreateRepository extends React.Component<
   private renderGitRepositoryWarning() {
     const isRepo = this.state.isRepository
 
-    if (this.state.path.length && !isRepo) {
+    if ((this.state.path.length && !isRepo) || this.state.path.length === 0) {
I've updated the PR. Good catch on the non-nullable string. It basically removes any possible edge cases.

**Comment by tierninho** [comment]
UI looks good. 

Simple tests passing:
- [x] Remove the path and no warning is present
- [x] Warning still present if a Git repo
- [x] link in warning is still active

File: app/src/ui/add-repository/create-repository.tsx (hunk 1 above)
Comment (iAmWillShepherd) [reply app/src/ui/add-repository/create-repository.tsx]: > It basically removes any possible edge cases.

Typescript will still let you assign `undefined` to a string and will accept `undefined` being passed as an argument to a function.

```ts
function f(a: string) {
  return a.length
}

f(undefined) //Cannot read property 'length' of undefined​​
```

☝️ TSC is happy to **not** call this an error and let it slide.

Comment (JakeHL) [reply app/src/ui/add-repository/create-repository.tsx]: Ah! okay - I'll implement a check for that case this evening 👍

**Comment by JakeHL** [comment]
Hi All - I've updated the condition once more. Hopefully this covers all bases. 👍

**Comment by shiftkey** [comment]
I'm not sure if this _needs_ to go out with 1.4.3 - had a chance to look over the latest changes @iAmWillShepherd?

**Comment by iAmWillShepherd** [comment]
I'll merge this after the next release, so it can get some beta time first.
